src/main/cpp/signal_handler.h

```diff
@@ -34,14 +35,12 @@ class SignalHandler {
 
     bool stopSigprof() { return updateSigprofInterval(0); }
 
-    ~SignalHandler() {
-        delete[] timingIntervals;
-    }
+    ~SignalHandler() {}
 
 private:
     int intervalIndex;
-    int *timingIntervals;
     int currentInterval;
+    std::array<int, NUMBER_OF_INTERVALS> timingIntervals;
```

Comment:
Should we not be deleting the `timingIntervals` std::array when the SignalHandler gets destroyed - looks like the SignalHandler owns the array.